### PR TITLE
chore: add some tests around the DockerConfig class

### PR DESCRIPTION
**Pull request by garethjevans**
This PR introduces unit tests for the 2 objects `DockerConfig` and `InfraConfig`, to ensure non regression about #177 and code quality improvement.

This PR is co-authored by @dduportal @MarckK and @garethjevans , and should be reviewed by another person than the 3 of us :)

### CODE REVIEW

File: test/groovy/DockerConfigTest.groovy [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import io.jenkins.infra.DockerConfig
+import io.jenkins.infra.InfraConfig
+
+import static org.junit.Assert.assertEquals
+import static org.junit.Assert.assertFalse
+import static org.junit.Assert.assertTrue
+
+class DockerConfigTest {
+
+    static String testImageName = "bitcoinMinerImage"
```
Comment (oleg-nenashev) [inline test/groovy/DockerConfigTest.groovy:12]: Maybe better to rename to prevent scanner tools from grumbling at some point

**Review by oleg-nenashev — DISMISSED** [review]

File: test/groovy/DockerConfigTest.groovy (hunk 1 above)
Comment (garethjevans) [reply test/groovy/DockerConfigTest.groovy]: done

**Review by oleg-nenashev — DISMISSED** [review]

File: test/groovy/DockerConfigTest.groovy [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    static String testImageName = "imagename"
+
+    @Test
+    void canHandleDefaultConfiguration() throws Exception {
+        def infraConfig = new InfraConfig(['JENKINS_URL':'https://ci.jenkins.io/'])
+        assertTrue(infraConfig.isRunningOnJenkinsInfra())
+        assertFalse(infraConfig.isInfra())
+        assertFalse(infraConfig.isTrusted())
+
+        def dockerConfig = new DockerConfig(testImageName, [:], infraConfig)
```
Comment (dduportal) [inline test/groovy/DockerConfigTest.groovy:21]: Proposal: change the constructor signature to have `config` as last argument

For instance:

```
public DockerConfig(String imageName, InfraConfig infraConfig, Map config=[:]) {
```

File: test/groovy/DockerConfigTest.groovy [hunk 3]
```diff
@@ -0,0 +1,92 @@
+
```
Comment (dduportal) [inline test/groovy/DockerConfigTest.groovy:1]:
```suggestion
package io.jenkins.infra
```

File: src/io/jenkins/infra/InfraConfig.groovy [hunk 4]
```diff
@@ -4,7 +4,7 @@ class InfraConfig implements Serializable {
   String jenkinsURL
 
   public InfraConfig(Object env) {
-    this.jenkinsURL = env.JENKINS_URL
+    this.jenkinsURL = env?.JENKINS_URL ?: ""
```
Comment (dduportal) [inline src/io/jenkins/infra/InfraConfig.groovy:7]: should use `''`

**Comment by garethjevans** [comment]
Hi @oleg-nenashev, would you be able to take a look at this again, thanks

**Review by timja — APPROVED** [review]

**Review by markyjackson-taulia — APPROVED** [review]

**Pull request merged**